tests/python/relax/test_codegen_cudnn.py

```diff
@@ -0,0 +1,287 @@
+    )
+
+
+@pytest.mark.parametrize(
+    "data_shape, weight_shape, dtype, with_bias, activation",
+    [
+        # Regular
+        ((16, 32, 32, 16), (32, 3, 3, 16), "float32", False, "none"),
+        # Bias
+        ((16, 32, 32, 16), (32, 3, 3, 16), "float32", True, "none"),
+        # Bias+ReLU
+        ((16, 32, 32, 16), (32, 3, 3, 16), "float32", True, "relu"),
+        # Bias+ReLU+half
+        ((16, 32, 32, 16), (32, 3, 3, 16), "float16", True, "relu"),
+    ],
+)
+def test_conv2d_offload(data_shape, weight_shape, dtype, with_bias, activation):
+    input = np.random.randn(*data_shape).astype(dtype)
+    weight = np.random.randn(*weight_shape).astype(dtype)
+
+    if with_bias:
+        oc = weight_shape[0]
+        bias = np.random.randn(1, 1, 1, oc).astype(dtype)
+        args = (input, weight, bias)
+    else:
+        bias = None
+        args = (input, weight)
+
+    activation = _activation_table[activation]
+
+    mod = get_relax_conv2d_module(
+        data_shape,
+        weight_shape,
+        dtype,
+        with_bias=with_bias,
+        activation=activation,
+    )
+
+    out = get_result_with_relax_cudnn_offload(mod, args)
+    ref = build_and_run(mod, args, "llvm", legalize=True)
+    if dtype == "float16":
+        tvm.testing.assert_allclose(out, ref, rtol=1e-1, atol=1e-1)
+    else:
+        tvm.testing.assert_allclose(out, ref, rtol=1e-2, atol=1e-2)
+
+
+@pytest.mark.parametrize(
+    "data_shape, weight_shape, dtype, with_bias, activation",
+    [
+        # Regular
+        ((16, 16, 32, 32), (32, 16, 3, 3), "float32", False, "none"),
+        # Bias
+        ((16, 16, 32, 32), (32, 16, 3, 3), "float32", True, "none"),
+        # Bias+ReLU
+        ((16, 16, 32, 32), (32, 16, 3, 3), "float32", True, "relu"),
+        # Bias+ReLU+half
+        ((16, 16, 32, 32), (32, 16, 3, 3), "float16", True, "relu"),
+    ],
+)
+def test_conv2d_nchw_oihw_offload(data_shape, weight_shape, dtype, with_bias, activation):
+    input = np.random.randn(*data_shape).astype(dtype)
+    weight = np.random.randn(*weight_shape).astype(dtype)
+
+    if with_bias:
+        oc = weight_shape[0]
+        bias = np.random.randn(1, oc, 1, 1).astype(dtype)
+        args = (input, weight, bias)
+    else:
+        bias = None
+        args = (input, weight)
+
+    activation = _activation_table[activation]
+
+    mod = get_relax_conv2d_module(
+        data_shape,
+        weight_shape,
+        dtype,
+        with_bias=with_bias,
+        activation=activation,
+        data_layout="NCHW",
+        kernel_layout="OIHW",
+    )
+
+    out = get_result_with_relax_cudnn_offload(mod, args)
+    ref = build_and_run(mod, args, "llvm", legalize=True)
+    if dtype == "float16":
+        tvm.testing.assert_allclose(out, ref, rtol=1e-1, atol=1e-1)
+    else:
+        tvm.testing.assert_allclose(out, ref, rtol=1e-2, atol=1e-2)
+
+
+def test_conv2d_cuda_graph():
```

Comment:
I mean, this test is probably not putting the conv2d into a graph. So just remove this entire test. In our cuda graph rewriting pass, we don't capture an op which depends on input variables (since those pointers might change).

Reply:
I see